src/Microsoft.Android.Build.BaseTasks/AndroidToolTask.cs

```diff
@@ -0,0 +1,24 @@
+using System;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.Android.Build.Tasks
+{
+	public abstract class AndroidToolTask : ToolTask
+	{
+		public abstract string TaskPrefix { get; }
+
+		public override bool Execute ()
+		{
+			try {
+				return RunTask ();
+			} catch (Exception ex) {
+				Log.LogUnhandledException (TaskPrefix, ex);
+				return false;
+			}
+		}
+
+		// Most ToolTask's do not override Execute and
+		// just expect the base to be called
+		public virtual bool RunTask () => base.Execute ();
```

Comment:
I'm not sure I understand how this "works" w/o causing stack overflow if the subclass doesn't "do" anything:

```csharp
class ExampleTask : AndroidToolTask {
    public override string TaskPrefix => "FOO";
}
```

Wouldn't `AndroidToolTask.Execute()` all `AndroidToolTask.RunTask()` call `.Execute()` call `.RunTask()` call…?

"Feels like" this should instead have `RunTask()` be `abstract`, and `Execute()` should be `override sealed`.

Reply:
This is:
* `AndroidToolTask.Execute()`
* `AndroidToolTask.RunTask()`
* `base.Execute()` or `ToolTask.Execute()` from MSBuild

Hopefully, should not be a stack overflow.

I don't think we can mark anything `sealed` as mentioned: https://github.com/xamarin/xamarin-android-tools/pull/101#discussion_r551219608